(FACT-1557) Extend the cloud.provider fact to Google Compute Engine

natemccurdy opened the pull request:
Similar to #2475 , this allows the `cloud.provider` fact to resolve on Google Compute Engine nodes to `gce`.

For example:
```
[user@a_gce_node]> facter cloud.provider
gce
```

This partially resolves https://tickets.puppetlabs.com/browse/FACT-1557

Note that I chose to have this resolve to the string "`gce`" because that's what the current `virtual` fact resolves to, so Facter/Puppet users are probably most familiar with "gce" meaning Google Compute Engine.

puppetlabs-jenkins commented:
Can one of the admins verify this patch?

joshcooper commented:
@natemccurdy could you prepend `(FACT-1557)` to your commit summary and force push to your feature branch?

natemccurdy commented:
> could you prepend `(FACT-1557)` to your commit summary and force push to your feature branch?
@joshcooper Done. Thanks for looking!

natemccurdy commented:
Offering given to the Rubocop gods, tests updated for lint failures, and branch force pushed. The rubocop tests should pass now.

natemccurdy commented:
Guess I didn't make the right offering, sorry RuboCop gods.
This time I fixed the lint issue, and verified locally with:
```
~/src/puppetlabs/facter [FACT-1557/cloud_provider_gce] be rubocop --parallel
Inspecting 1730 files
...............................................................................
...............................................................................
...............................................................................
...............................................................................
...............................................................................
...............................................................................

1730 files inspected, no offenses detected
```

natemccurdy commented:
@joshcooper  What do you recommend I do about the new RuboCop warnings?

```
New offenses:
  - lib/facter/facts/linux/cloud/provider.rb:
    - Metrics/CyclomaticComplexity: changed from 0 to 1
  - lib/facter/facts/windows/cloud/provider.rb:
    - Metrics/CyclomaticComplexity: changed from 0 to 1
```

joshcooper commented:
@natemccurdy could you add [exclusions for those files to `.rubocop.yml`](https://github.com/puppetlabs/facter/blob/366e6f9e8380c51a08a23bde2c0c96520e5d7610/.rubocop.yml#L83-L93) and add that as a second commit? If the rubocop checks are performed for each commit, you may need to switch the order so the files are excluded first?

natemccurdy commented:
> ...add [exclusions for those files to `.rubocop.yml`

Done 👍

joshcooper reviewed (approved):
Verified this behaves as expected on GCE:

```
~/facter# bundle exec facter cloud.provider
gce
```

and it has parity with other cloud  providers facts like https://tickets.puppetlabs.com/browse/FACT-1441
